doc/api.rst

```diff
@@ -0,0 +1,234 @@
+Introduction
+++++++++++++
+
+The dl_esm_inf (for Daresbury Laboratory Earth-System Modelling
+Infrastructure) library provides basic support for finite-difference
+earth-system-type models written in Fortran. It currently
+supports two-dimensional, finite-difference models.
+
+The first version of this library was developed to support 2D finite-
+difference shallow-water models in the GOcean Project.
+
+
+.. _gocean1.0-grid:
+
+Grid
+++++
+
+The dl_esm_inf library contains a ``grid_mod`` module which defines a
+``grid_type`` and associated constructor::
+
+  use grid_mod
+  ...
+  !> The grid on which our fields are defined
+  type(grid_type), target :: model_grid
+  ...
+  ! Create the model grid
+  model_grid = grid_type(GO_ARAKAWA_C,                                 &
+                         (/GO_BC_EXTERNAL,GO_BC_EXTERNAL,GO_BC_NONE/), &
+                         GO_OFFSET_NE)
+
+.. note::
+  The grid object itself must be declared with the ``target``
+  attribute. This is because each field object will contain a pointer to
+  it.
+
+The ``grid_type`` constructor takes three arguments:
+
+ 1. The type of grid (only GO_ARAKAWA_C is currently supported)
+ 2. The boundary conditions on the domain for the *x*, *y* and *z* dimensions (see below). The value for the *z* dimension is currently ignored.
+ 3. The 'index offset' - the convention used for indexing into offset fields.
+
+Three types of boundary condition are currently supported:
+
+.. tabularcolumns:: |l|L|
+
+===============  =========================================
+Name             Description
+===============  =========================================
+GO_BC_NONE       No boundary conditions are applied.
+GO_BC_EXTERNAL   Some external forcing is applied. This must be implemented by a kernel. The domain must be defined with a T-point mask (see :ref:`gocean1.0-grid-init`).
+GO_BC_PERIODIC   Periodic boundary conditions are applied.
+===============  =========================================
+
+The infrastructure requires this information in order to determine the
+extent of the model grid.
+
+The index offset is required because a model (kernel) developer has
+choice in how they actually implement the staggering of variables on a
+grid. This comes down to a choice of which grid points in the vicinity
+of a given T point have the same array (*i*, *j*) indices. In
+the diagram below, the image on the left corresponds to choosing those
+points to the South and West of a T point to have the same (*i*, *j*)
+index. That on the right corresponds to choosing those points to the
```

Comment:
"That on" sounds odd to me: "The one" ?? I would have just changed it myself, but thought I'll better ask you for things regarding English language :)

Reply:
:-) "That on the right" means "the image on the right" and follows from the previous sentence that refers to "the image on the left". I *think* it's correct but I could be wrong...

Reply:
Who am I to argue English with a native speaker ;)